buildifier-wrapper.sh

```diff
@@ -56,15 +58,9 @@ fi
 
 if echo "$sha  $tmp_binary" | $shabin --check --status; then
   chmod +x "$tmp_binary"
-
-  # Protect against races of concurrent hooks downloading the same binary
-  if [[ ! -x "$binary" ]]; then
-    mv "$tmp_binary" "$binary"
-  fi
-
+  mv "$tmp_binary" "$binary"
   exec "$binary" "$@"
 else
   echo "error: buildifier sha mismatch" >&2
-  rm -f "$binary"
```

Comment:
I think we still want to keep this removal? this is the case the binary is untrusted

Reply:
Sure, reverted this part. Thanks for taking a look!

Reply:
Emmmm, on second thought, it’s probably best to not `rm -f "$binary"`. Consider the following (unlikely but possible) scenario:

1. The wrapper is called when there is no existing binary.
2. While the binary is being downloaded, the wrapper is called again, which downloads the binary again (to a separate tmp file).
3. The first download succeeds and becomes $binary.
4. The second download fails. If we keep `rm -f "$binary"`, the good binary downloaded from the first run will be deleted, which leads to an unnecessary re-download in the future. The bad tmp binary from the second download will be removed on exit anyways, because of `trap 'rm -f "$tmp_binary"' EXIT`.

Reply:
++ removing tmp_binary is all that matters, sgtm